Feature/17 remove isample

GegznaV opened the pull request:
Closes #17

GegznaV committed:
Implement `sample(..., index = TRUE)`

GegznaV committed:
Deprecate `isample()`

GegznaV committed:
Update DEPRECATED-isample.R

GegznaV committed:
Fix unit tests

GegznaV committed:
Update vignette

GegznaV committed:
Update NEWS.md

GegznaV committed:
Fix deprecated function

GegznaV committed:
Update unit tests

GegznaV committed:
Update sample.R

GegznaV committed:
Fix documentation

GegznaV committed:
Update sample.R

GegznaV committed:
Update DEPRECATED-isample.R

GegznaV committed:
Update documentation

GegznaV committed:
Update documentation

GegznaV committed:
Update documentation

GegznaV committed:
Change order of arguments

GegznaV committed:
Update documentation: title and return section

GegznaV committed:
Update unit tests

bryanhanson reviewed (commented):
The only thing I see is
`#'        to choose. Id missing, defaults to number of rows in the object:`
has a typo, "Id" should be "If".

OK to merge.

GegznaV committed:
Fix documentation

Addresses https://github.com/r-hyperspec/hyperSpec/pull/53#pullrequestreview-728808932
